cosmology/GravityFieldContainer.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef IPPL_FIELD_CONTAINER_H
+#define IPPL_FIELD_CONTAINER_H
+
+#include <memory>
+
+#include "Manager/BaseManager.h"
+
+// Define the FieldsContainer class
+/**
+ * @brief A container class that manages various fields, mesh, and field layouts
+ *
+ * @tparam T Data type used for the fields (e.g., double)
+ * @tparam Dim Dimensionality of the container (default is 3)
+ */
+template <typename T, unsigned Dim = 3>
+class FieldContainer{
+public:
+
+    /**
+     * @brief Constructor for the FieldContainer class
+     * 
+     * @param hr Grid spacing vector
+     * @param rmin Minimum range vector
+     * @param rmax Maximum range vector
+     * @param decomp Decomposition array indicating which dimensions are decomposed
+     * @param domain The domain over which the mesh is defined
+     * @param origin The origin of the mesh
+     * @param isAllPeriodic Flag indicating if all dimensions are periodic
+     */
+    FieldContainer(Vector_t<T, Dim>& hr, Vector_t<T, Dim>& rmin,
+                   Vector_t<T, Dim>& rmax, std::array<bool, Dim> decomp,
+                   ippl::NDIndex<Dim> domain, Vector_t<T, Dim> origin,
+                   bool isAllPeriodic)
+        : hr_m(hr)
+        , rmin_m(rmin)
+        , rmax_m(rmax)
+        , decomp_m(decomp)
+        , mesh_m(domain, hr, origin)
+        , fl_m(MPI_COMM_WORLD, domain, decomp, isAllPeriodic) {}
+    /**
+     * @brief Destructor for the FieldContainer class
+     */
+    ~FieldContainer(){}
+
+
+
+    ///< Get the vector field F
+    VField_t<T, Dim>& getF() { return F_m; }    
+    ///< Set the vector field F 
+    void setF(VField_t<T, Dim>& F) { F_m = F; }   
+
+    ///< Get the scalar field rho 
+    Field_t<Dim>& getRho() { return rho_m; }     
+    ///< Set the scalar field rho
+    void setRho(Field_t<Dim>& rho) { rho_m = rho; }
```

Comment:
I would be more specific about the physical meaning i.e. "Get/Set the scalar field rho (mass density)"